### PR TITLE
[NEW] Provide new Livechat Api setCustomFields

### PR DESCRIPTION
**Pull request by lolimay**
# What? :boat:

Provide a new Livechat Api by `modify.getUpdater().getLivechatUpdater().setCustomFields`
# Why? :thinking:

# Links :earth_americas:

https://app.clickup.com/t/24zmb4

# PS :eyes:
PR in the RC side: https://github.com/RocketChat/Rocket.Chat/pull/18912

### CODE REVIEW

**Comment by lolimay** [comment]
image

@renatobecker This should be working now, can you please help check?

**Comment by lolimay** [comment]
```js
import { ILivechatUpdater } from '../../definition/accessors';
import { ILivechatRoom, ILivechatTransferData, IVisitor } from '../../definition/livechat';
import { AppBridges } from '../bridges';

export class LivechatUpdater implements ILivechatUpdater {
    constructor(private readonly bridges: AppBridges, private readonly appId: string) { }

    public transferVisitor(visitor: IVisitor, transferData: ILivechatTransferData): Promise<boolean> {
        return this.bridges.getLivechatBridge().transferVisitor(visitor, transferData, this.appId);
    }

    public closeRoom(room: ILivechatRoom, comment: string): Promise<boolean> {
        return this.bridges.getLivechatBridge().closeRoom(room, comment, this.appId);
    }

    public setCustomFields(token: IVisitor['token'], key: string, value: string, overwrite: boolean): Promise<any> {
        return this.bridges.getLivechatBridge().setCustomFields({ token, key, value, overwrite }, this.appId);
    }
}
```
The app I used for testing.

**Comment by d-gubert** [comment]
This requires changes on the Rocket.Chat side as well, right?

File: src/server/bridges/ILivechatBridge.ts [hunk 1]
```diff
@@ -27,4 +27,6 @@ export interface ILivechatBridge {
     closeRoom(room: ILivechatRoom, comment: string, appId: string): Promise<boolean>;
     findRooms(visitor: IVisitor, departmentId: string | null, appId: string): Promise<Array<ILivechatRoom>>;
     findDepartmentByIdOrName(value: string, appdId: string): Promise<IDepartment | undefined>;
+
+    setCustomFields(data: { token: IVisitor['token']; key: string; value: string; overwrite: boolean }, appId: string): Promise<any>;
```
Comment (d-gubert) [inline src/server/bridges/ILivechatBridge.ts:31]: Is it possible for us to determine what should be the return of the method by its implementation on Rocket.Chat? We really should avoid `any` type annotations in the APIs

**Review by d-gubert — CHANGES_REQUESTED** [review]

File: src/server/bridges/ILivechatBridge.ts (hunk 1 above)
Comment (lolimay) [reply src/server/bridges/ILivechatBridge.ts]: The setCustomFields originally returns **the number of documents affected while performing a `update` operation to the db**. I made some preprocess here: https://github.com/RocketChat/Rocket.Chat.Apps-engine/pull/313/commits/85cf62cd3994dc96127275cc60455706eacc5d82#diff-f859ad2ae3aa3b6f9d373e0f42479597R18 to make its return type to `Promise<boolean>`. Does this preprocess make sense or not? Or

Comment (thassiov) [reply src/server/bridges/ILivechatBridge.ts]: IMO having a boolean as the return type makes the method simpler to reason about (and you already did it, right?)

Comment (lolimay) [reply src/server/bridges/ILivechatBridge.ts]: yeah, I already changed the signature to `....: Promise<boolean>`.

**Comment by thassiov** [comment]
LGTM

**Review by d-gubert — APPROVED** [review]

**Pull request merged**